src/Workspaces/Core/Portable/LanguageServices/SemanticsFactsService/ISemanticFactsService.cs

```diff
@@ -111,6 +112,10 @@ internal interface ISemanticFactsService : ILanguageService
             SemanticModel semanticModel, SyntaxNode location,
             SyntaxNode containerOpt, string baseName, IEnumerable<string> usedNames, CancellationToken cancellationToken);
 
+        SyntaxToken GenerateUniqueName(
```

Comment:
GenerateUniqueLocalName actually also looks at fields and properties, so I had to add an interface method to allow me to pass in a custom filter.